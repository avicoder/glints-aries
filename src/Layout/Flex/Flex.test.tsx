import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import { Flex } from './index';

describe('<Flex /> spacing properties', () => {
  it('should match snapshot when margin is set', () => {
    const { asFragment } = render(<Flex m={64} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when all margin directions are set', () => {
    const { asFragment } = render(<Flex mt={64} mr={64} mb={64} ml={64} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when padding is set', () => {
    const { asFragment } = render(<Flex p={64} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when all padding directions are set', () => {
    const { asFragment } = render(<Flex pt={64} pr={64} pb={64} pl={64} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when responsive margin is set', () => {
    const { asFragment } = render(<Flex m={{ default: 32, desktopS: 64 }} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when responsive padding is set', () => {
    const { asFragment } = render(<Flex p={{ default: 32, desktopS: 64 }} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('<Flex /> flex properties', () => {
  it('should match snapshot when flexDirection is set', () => {
    const { asFragment } = render(<Flex flexDirection="column" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when justifyContent is set', () => {
    const { asFragment } = render(<Flex justifyContent="center" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when alignItems is set', () => {
    const { asFragment } = render(<Flex alignItems="center" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when flexWrap is set', () => {
    const { asFragment } = render(<Flex flexWrap="wrap" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
