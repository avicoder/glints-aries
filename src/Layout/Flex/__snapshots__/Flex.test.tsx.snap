// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> flex properties should match snapshot when alignItems is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 chpOLw Flex__StyledFlex-sc-11ryoct-0 hOHBUN"
  />
</DocumentFragment>
`;

exports[`<Flex /> flex properties should match snapshot when flexDirection is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 chpOLw Flex__StyledFlex-sc-11ryoct-0 heKDkF"
  />
</DocumentFragment>
`;

exports[`<Flex /> flex properties should match snapshot when flexWrap is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 chpOLw Flex__StyledFlex-sc-11ryoct-0 ffQToI"
  />
</DocumentFragment>
`;

exports[`<Flex /> flex properties should match snapshot when justifyContent is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 chpOLw Flex__StyledFlex-sc-11ryoct-0 bkexZU"
  />
</DocumentFragment>
`;

exports[`<Flex /> spacing properties should match snapshot when all margin directions are set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 ebktHH Flex__StyledFlex-sc-11ryoct-0 jfpAHx"
  />
</DocumentFragment>
`;

exports[`<Flex /> spacing properties should match snapshot when all padding directions are set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 jWilAT Flex__StyledFlex-sc-11ryoct-0 jfpAHx"
  />
</DocumentFragment>
`;

exports[`<Flex /> spacing properties should match snapshot when margin is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 cMvEap Flex__StyledFlex-sc-11ryoct-0 jfpAHx"
  />
</DocumentFragment>
`;

exports[`<Flex /> spacing properties should match snapshot when padding is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 hA-dxnS Flex__StyledFlex-sc-11ryoct-0 jfpAHx"
  />
</DocumentFragment>
`;

exports[`<Flex /> spacing properties should match snapshot when responsive margin is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 cYQtfv Flex__StyledFlex-sc-11ryoct-0 jfpAHx"
  />
</DocumentFragment>
`;

exports[`<Flex /> spacing properties should match snapshot when responsive padding is set 1`] = `
<DocumentFragment>
  <div
    class="Box__StyledBox-pr7b7l-0 iVdNyz Flex__StyledFlex-sc-11ryoct-0 jfpAHx"
  />
</DocumentFragment>
`;
